### PR TITLE
core: arm64: use adr_l to reference stack_tmp_stride

### PR DESCRIPTION
**Pull request by jforissier**
When CFG_NUM_THREADS and/or CFG_TEE_CORE_NB_CORE become large enough,
link errors are reported:

 $ make -s CFG_TEE_CORE_NB_CORE=48 CFG_NUM_THREADS=48 \
           PLATFORM=vexpress-qemu_armv8a
 out/arm-plat-vexpress/core/arch/arm/kernel/entry_a64.o: in function `clear_bss':
 core/arch/arm/kernel/entry_a64.S:160:(.text._start+0x98): relocation truncated to fit: R_AARCH64_ADR_PREL_LO21 against symbol `stack_tmp_stride' defined in .identity_map.stack_tmp_stride section in out/arm-plat-vexpress/core/arch/arm/kernel/thread.o
 out/arm-plat-vexpress/core/arch/arm/kernel/entry_a64.o: in function `cpu_on_handler':
 core/arch/arm/kernel/entry_a64.S:443:(.text.cpu_on_handler+0x50): relocation truncated to fit: R_AARCH64_ADR_PREL_LO21 against symbol `stack_tmp_stride' defined in .identity_map.stack_tmp_stride section in out/arm-plat-vexpress/core/arch/arm/kernel/thread.o

~~Fix the issue by replacing the addr instruction with the pattern:
 addrp x, sym         ; load page address of sym
 add x, x, :lo12:sym  ; add offset of sym in page~~
Fix the issue by replacing the addr instruction with the adr_l macro.

Signed-off-by: Jerome Forissier <jerome@forissier.org>

### CODE REVIEW

File: core/arch/arm/kernel/entry_a64.S [hunk 1]
```diff
@@ -25,7 +25,8 @@
 		cmp	x0, #CFG_TEE_CORE_NB_CORE
 		/* Unsupported CPU, park it before it breaks something */
 		bge	unhandled_cpu
-		adr	x1, stack_tmp_stride
+		adrp	x1, stack_tmp_stride
```
Comment (jenswi-linaro) [inline core/arch/arm/kernel/entry_a64.S:28]: Please use `adr_l` instead.

Comment (jforissier) [reply core/arch/arm/kernel/entry_a64.S:28]: Sure. I was mislead by the following lines (`stack_tmp_export`) which still use `adrp`/`add` :-/ I shall convert them first.

**Comment by jforissier** [comment]
Update.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**